src/cript/api/utils/save_helper.py

```diff
@@ -1,29 +1,93 @@
+    suppress_attributes: Dict[str, Set[str]] = field(default_factory=dict)
+
+    def __add__(self, other: "_InternalSaveValues") -> "_InternalSaveValues":
+        """
+        Implement a short hand to combine two of these save values, with `+`.
+        This unions, the `saved_uuid`.
+        And safely unions `suppress_attributes` too.
+        """
+        # Make a manual copy of `self`.
+        return_value = _InternalSaveValues(self.saved_uuid.union(other.saved_uuid), self.suppress_attributes)
+
+        # Union the dictionary.
+        for uuid in other.suppress_attributes:
+            try:
+                # If the uuid exists in both `suppress_attributes` union the value sets
+                return_value.suppress_attributes[uuid] = return_value.suppress_attributes[uuid].union(other.suppress_attributes[uuid])
+            except KeyError:
+                # If it only exists in one, just copy the set into the new one.
+                return_value.suppress_attributes[uuid] = other.suppress_attributes[uuid]
+        return return_value
+
+
+def _fix_node_save(api, node, response, save_values: _InternalSaveValues) -> _InternalSaveValues:
     """
     Helper function, that attempts to fix a bad node.
     And if it is fixable, we resave the entire node.
 
     Returns set of known uuids, if fixable, otherwise False.
     """
-    assert response["code"] in (400, 409)
+    if response["code"] not in (400, 409):
+        raise RuntimeError(f"The internal helper function `_fix_node_save` has been called for an error that is not yet implemented to be handled {response}.")
+
     if response["error"].startswith("Bad uuid:") or response["error"].strip().startswith("Duplicate uuid:"):
-        missing_uuid = get_uuid_from_error_message(response["error"])
+        missing_uuid = _get_uuid_from_error_message(response["error"])
         missing_node = find_node_by_uuid(node, missing_uuid)
 
         # Now we save the bad node extra.
         # So it will be known when we attempt to save the graph again.
         # Since we pre-saved this node, we want it to be UUID edge only the next JSON.
         # So we add it to the list of known nodes
-        known_uuid.union(api._internal_save(missing_node, known_uuid))  # type: ignore
+        returned_save_values = api._internal_save(missing_node, save_values)
+        save_values += returned_save_values
         # The missing node, is now known to the API
-        known_uuid.add(missing_uuid)
+        save_values.saved_uuid.add(missing_uuid)
         # Recursive call.
         # Since we should have fixed the "Bad UUID" now, we can try to save the node again
-        return api._internal_save(node, known_uuid)
-    return False
+        return api._internal_save(node, save_values)
+
+    # Handle all duplicate items warnings if possible
+    if response["error"].startswith("duplicate item"):
+        for search_dict_str in re.findall(r"\{(.*?)\}", response["error"]):
```

Comment:
please put an explanation of the regex because they are typically hard to read and understand

Reply:
I added a comment.

Reply:
I think this might need more comments because it could be hard to follow the logic, something like

```
# find all JSON blocks that have `error` in them
for search_dict_str in re.findall(r"\{(.*?)\}", response["error"]):
```

Reply:
I have added a comment for this regex.